src/rtctools/optimization/multi_seed_mixin.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, Dict, Iterable
+
+from rtctools.optimization.optimization_problem import OptimizationProblem
+from rtctools.optimization.timeseries import Timeseries
+
+
+class MultiSeedMixin(OptimizationProblem):
+    """
+    Enables a workflow to solve an optimization problem by trying multiple seeds.
+    """
+
+    def __init__(self, **kwargs):
+        self.__selected_seed_id = None
+        super().__init__(**kwargs)
+
+    def use_seed_id(self) -> bool:
```

Comment:
Is this method needed?

For goal programming, couldn't the logic be simplified to:
- if preprocessing is True, select the seed from the seed IDs
- if preprocessing is False, use the seed() implementation from the superclass

Reply:
I'm not sure if I understand what you mean. When calling `super().optimize` from `MultiSeedMixin`, the argument `preprocessing` will always be set to `False`. We somehow need to determine if we want to use an initial seed (e.g. when solving for the first priority) or a seed based on the last results (e.g. when solving for the next priorities).

Reply:
By default, optimize() is called with preprocessing set to True. It's only in GoalProgrammingMixin that the method is overridden with an implementation that calls optimize() multiple times with preprocessing=False. You can use this to override the top-level implementation with a preprocessing=True/False check.

Reply:
The way `MultiSeedMixin.optimize()` is now implemented is that it will call pre at the start if preprocessing is True and within a loop for each seed call `super().optimize` with preprocessing is False. If MultiSeedMixin is the top level, it has no idea if GoalProgrammingMixin is at its first run or not, so we leave this logic up to the user by allowing him to override this method.

Reply:
I have to admit that during development, I had in mind that `MultiSeedMixin` would be the top-level class in the inheritance hierarchy although we never impose this. Perhaps by making it sub-level to HomotopyMixin/GoalProgrammingMixin the logic can indeed be simplified. Let me check this.